tests/test_components.py

```diff
@@ -481,37 +541,80 @@ def test_parse(self):
 
 
 class CollectionsTestCase(TestCase):
-    def test_not_enabled(self):
-        component = Collections(Mock())
-        component.raw_data = {"enable": False}
-        component.parse()
-        self.assertFalse(ModerationCollection.objects.exists())
-
-    def test_enabled(self):
-        user1 = UserFactory()
-        user2 = UserFactory()
-        role1 = Role.objects.create(name="Role 1", user=user1)
-        role2 = Role.objects.create(name="Role 2", user=user2)
-        versions = PageVersionFactory.create_batch(8)
-        pages = [version.content.page for version in versions]
-        workflow = Workflow.objects.create(name="Workflow 1", is_default=False)
-        workflow.steps.create(role=role1, order=1)
-        workflow.steps.create(role=role2, order=2)
-        bootstrap = Mock(
-            users={"moderator": user1, "moderator2": user1, "moderator3": user1},
-            pages={
-                "page1": pages[0],
-                "page2": pages[1],
-                "page3": pages[2],
-                "page4": pages[3],
-                "page5": pages[4],
-                "page6": pages[5],
-                "page7": pages[6],
-                "page8": pages[7],
-            },
-            workflows={"wf4": workflow, "wf5": workflow},
-        )
-        component = Collections(bootstrap)
-        component.raw_data = {"enable": True}
+    def setUp(self):
+        self.user = UserFactory(username="user1")
+        self.wf1 = Workflow.objects.create(name="Workflow 1")
```

Comment:
we should create this using factory as well.

Reply:
I don't understand. User is created with Factory. Do you mean Workflow? If so I don't think this is necessary because the factory wouldn't do anything special and also this hasn't been followed for other objects in previous tests such as Role.

Reply:
Yes, I refer to workflow as commented on that line. We should start creating Factory as it, not a valid argument saying we were not doing before means we shouldn't.